test/hotspot/jtreg/runtime/NMT/SummarySanityCheck.java

```diff
@@ -85,6 +85,15 @@ public static void main(String args[]) throws Exception {
           long typeCommitted = Long.parseLong(typeMatcher.group("committed"));
           long typeReserved = Long.parseLong(typeMatcher.group("reserved"));
 
+          if(typeMatcher.group("readonly") != null) {
+            long typeReadOnly = Long.parseLong(typeMatcher.group("readonly"));
```

Comment:
Add a space behind the "if".

Also, as @tstuefe suggested, if "readonly" is present in the outline, check that the line also contains "Shared class space", since no other lines should contain "readonly".